cmd/hashObject.go

```diff
@@ -1,27 +1,70 @@
 /*
 Copyright © 2024 NAME HERE <EMAIL ADDRESS>
-
 */
 package cmd
 
 import (
 	"fmt"
+	"gogitty/internal/core"
+	"io/fs"
+	"os"
+	"path/filepath"
 
 	"github.com/spf13/cobra"
 )
 
 // hashObjectCmd represents the hashObject command
 var hashObjectCmd = &cobra.Command{
-	Use:   "hashObject",
+	Use:   "hash",
 	Short: "A brief description of your command",
-	Long: `A longer description that spans multiple lines and likely contains examples
-and usage of using your command. For example:
-
-Cobra is a CLI library for Go that empowers applications.
-This application is a tool to generate the needed files
-to quickly create a Cobra application.`,
+	Long:  `converts an existing file into a git object`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("hashObject called")
+		var fp string
+		var stats fs.FileInfo
+		var err error
+
+		// check if custom path is provided or get current working directory
+		if len(args) > 0 {
+			fp = args[0]
+			stats, err = os.Stat(fp) // Capture error from os.Stat
```

Comment:
os.Stat returns path error which is a struct. Logging the object of err directly will serialize a struct which is not a good practice.

Ref - https://pkg.go.dev/io/fs#PathError